ucte/ucte-network/src/main/java/com/powsybl/ucte/network/UcteCountryCode.java

```diff
@@ -121,4 +125,15 @@ public static boolean isUcteCountryCode(char character) {
         }
     }
 
+    public static UcteCountryCode fromVoltagelevel(VoltageLevel voltageLevel) {
+        Country country = voltageLevel.getSubstation()
+                .flatMap(Substation::getCountry)
+                .orElseThrow(() -> new UcteException("No UCTE country found for substation"));
+        try {
+            return UcteCountryCode.valueOf(country.name());
+        } catch (IllegalArgumentException e) {
+            throw new UcteException(String.format("No UCTE country found for %s",country.name()));
```

Comment:
To fix the checkstyle issue:

```suggestion
            throw new UcteException(String.format("No UCTE country found for %s", country.name()));
```